hs_core/models.py

```diff
@@ -2119,12 +2161,16 @@ def check_irods_files(self, stop_on_error=False, log_errors=True,
                     errors.extend(error2)
                     ecount += ecount2
 
-            # Step 1: does every file here refer to an existing file in iRODS?
+            # Step 2: does every file here refer to an existing file in iRODS?
             for f in self.files.all():
                 if not istorage.exists(f.storage_path):
                     ecount += 1
                     msg = "check_irods_files: django file {} does not exist in iRODS"\
                         .format(f.storage_path)
+                    if clean_django:
+                        delete_resource_file(self.short_id, f.short_path, self.creator,
+                                             delete_logical_file=False)
```

Comment:
@hyi: @pkdash and I discussed this. We decided that 
1. this code can leave a logical file around. 
2. we need to update the code so that this won't happen. 
3. we will defer this until 1.11.0 because users won't see it at all.